app/src/main/java/com/sdpteam/connectout/event/EventCreatorViewModel.java

```diff
@@ -0,0 +1,26 @@
+package com.sdpteam.connectout.event;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModel;
+
+public class EventCreatorViewModel extends ViewModel {
+    private final EventDataManager model;
+
+
+    public EventCreatorViewModel(EventDataManager model) {
+        this.model = model;
+    }
+
+
+    public LiveData<Event> getValue(String eid) {
+        return model.getValue(eid);
+    }
+
+    public LiveData<Event> getValue(String uid, String title) {
+        return model.getValue(uid, title);
+    }
+
+    public boolean saveValue(Event event) {
+        return model.saveValue(event);
+    }
```

Comment:
I feel like you should deal with the boolean return value in the ViewModel rather than the Activity, but that's just a tip

Reply:
I agree, as it will be useful only for caching and offline mode, only the view model should handle a false returned value.
it is now fixed.